### PR TITLE
Allow passwordless accounts to be created

### PR DESCRIPTION
**Pull request by doomspork**
Existing processes within the company rely on staff being able to create
a user account with no password, later set with password reset. This
accounts will only be created via the API and UI in the administrative
portal.

### CODE REVIEW

File: lib/recognizer_web/controllers/accounts/api/user_registration_controller.ex [hunk 1]
```diff
@@ -0,0 +1,19 @@
+defmodule RecognizerWeb.Accounts.Api.UserRegistrationController do
```
Comment (btkostner) [inline lib/recognizer_web/controllers/accounts/api/user_registration_controller.ex:1]: I'd check to make sure this endpoint is only runnable by admins

File: lib/recognizer/accounts.ex [hunk 2]
```diff
@@ -162,9 +162,9 @@ defmodule Recognizer.Accounts do
       {:error, %Ecto.Changeset{}}
 
   """
-  def register_user(attrs) do
+  def register_user(attrs, opts \\ []) do
```
Comment (btkostner) [inline lib/recognizer/accounts.ex:165]: Sweet, this is pretty much exactly what I was thinking. I'd also pass the options down to `maybe_notify_new_user/1` so we can disable that too.

**Review by btkostner — CHANGES_REQUESTED** [review]

File: lib/recognizer/accounts.ex (hunk 2 above)
Comment (doomspork) [reply lib/recognizer/accounts.ex:165]: I'd think we still want the welcome email to send. I will confirm with sales that is what they expect.

Comment (btkostner) [reply lib/recognizer/accounts.ex:165]: I think it's pretty weird that we would send them an email and immediately require them to reset their password to access it. I'd at least change the wording to it

File: lib/recognizer_web/controllers/accounts/api/user_registration_controller.ex [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  alias RecognizerWeb.ErrorView
+
+  def create(conn, %{"user" => user_params}) do
+    with true <- staff_request?(conn),
+         {:ok, user} <- Accounts.register_user(user_params, skip_password: true) do
+      render(conn, "show.json", user: user)
+    end
+  end
+
+  def staff_request(conn) do
```
Comment (btkostner) [inline lib/recognizer_web/controllers/accounts/api/user_registration_controller.ex:15]:
```suggestion
  def staff_request?(conn) do
```

**Review by btkostner — APPROVED** [review]
once tests pass, this looks fine

File: lib/recognizer/accounts.ex (hunk 2 above)
Comment (doomspork) [reply lib/recognizer/accounts.ex:165]: I don't think they'll be immediately doing anything with their account, which is why we're creating accounts for them. The welcome email is the first opportunity Marketing has to engage the customer.

**Review by btkostner — APPROVED** [review]

**Review by btkostner — APPROVED** [review]

**Pull request merged**